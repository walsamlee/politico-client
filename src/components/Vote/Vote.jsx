/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';

class Vote extends Component {
    voteHandler = (event) => {
      event.preventDefault();
      const { voteData } = this.props;
      const token = localStorage.getItem('token');

      this.props.castVote(voteData, token);
    }

    render() {
      return (
        <div className="vote-date">
          <button type="button" className="btn btn-cart" onClick={this.voteHandler}>Vote</button>
        </div>
      );
    }
}

export default Vote;
