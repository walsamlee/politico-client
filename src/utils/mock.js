const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
