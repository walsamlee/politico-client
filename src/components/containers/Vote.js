import { connect } from 'react-redux';
import voteAction from '../../actions/voteAction';
import Vote from '../Vote/Vote';

const mapStateToProps = state => ({
  data: state,
});

const mapDispatchToProps = dispatch => ({
  castVote: (voteDate, token) => {
    dispatch(voteAction.castVote(voteDate, token));
  },
});

const vote = connect(mapStateToProps, mapDispatchToProps)(Vote);

export default vote;
