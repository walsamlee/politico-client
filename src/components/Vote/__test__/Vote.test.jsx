import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import Vote from '../Vote';

const { voteData } = mock;

describe('<Vote />', () => {
  it('should render vote component', () => {
    const wrapper = shallow(
      <Vote voteData={voteData} castVote={jest.fn()} />,
    );
    const event = {
      preventDefault: jest.fn(),
    };
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
    wrapper.instance().voteHandler(event);
  });
});
