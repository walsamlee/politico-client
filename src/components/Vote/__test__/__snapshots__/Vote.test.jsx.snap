// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vote /> should render vote component 1`] = `ShallowWrapper {}`;
